tests/42tags.pl

```diff
@@ -378,3 +385,49 @@ sub check_private_user_data {
          Future->done( 1 );
       });
    };
+
+test "Deleted tags appear in an incremental v2 /sync",
+   requires => [qw( first_api_client can_add_tag can_remove_tag can_sync )],
+
+   do => sub {
+      my ( $http ) = @_;
+
+      my ( $user, $room_id, $filter_id, $next_batch );
+
+      my $filter = {};
+
+      matrix_register_user_with_filter( $http, $filter )->then( sub {
+         ( $user, $filter_id ) = @_;
+
+         matrix_create_room( $user );
+      })->then( sub {
+         ( $room_id ) = @_;
+
+         matrix_sync( $user, $filter => $filter_id );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next_batch = $body->{next_batch};
+
+         matrix_add_tag( $user, $room_id, "test_tag", { order => 1 } );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id, since => $next_batch );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next_batch = $body->{next_batch};
+
+         matrix_remove_tag( $user, $room_id, "test_tag" );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id, since => $next_batch );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         my $room = $body->{rooms}{joined}{$room_id};
+         require_json_keys( $room, qw( private_user_data ) );
+
+         check_private_user_data( $room->{private_user_data}{events}, 1 );
```

Comment:
Reading this, it's really unclear what the 1 means.

Can you make it more clear?